subsys/net/ip/dhcpv4.c

```diff
@@ -881,13 +896,18 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			break;
 		}
 		default:
-			NET_DBG("option unknown: %d", type);
+			if (unhandled) {
+				NET_DBG("option unknown: %d", type);
+			}
 
 			if (net_pkt_skip(pkt, length)) {
-				NET_DBG("option unknown, skip err");
+				if (unhandled) {
+					NET_DBG("option unknown, skip err");
+				} else {
+					NET_DBG("option callback, skip err");
+				}
```

Comment:
IMO this If here is not needed, the log is just reporting `net_pkt_skip()` error.